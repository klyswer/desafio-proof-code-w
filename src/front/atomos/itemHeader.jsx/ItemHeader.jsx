import React from 'react';
import PropTypes from 'prop-types';
import Contador from './Contador';
import './style.css';

const ItemHeader = ({icono,title,action,cssCont,cssBtn,tipoCart}) => {
  return (
    <div className={`nav-item ${cssCont!==''?cssCont:''}`}>
      <button type="button" className={`nav-link default_btn_header ${cssBtn!==''?cssBtn:''}`} onClick={action}>
        {
          tipoCart === false ?(
            <div className="contenido_btn">
              <span>{icono}</span> 
              {title}
            </div>
          ):(
            <div className="d-flex justify-content-center aling-items-center">            
              <img height="25px" className="d-inline-block align-top" src={`${process.env.PUBLIC_URL}/images/lider-cart.svg`} alt="Lider-icon" />
              <Contador number={10} />
            </div>
          )
        }
      </button>
    </div>         
  );
};


ItemHeader.defaultProps = {
  icono:null,
  title:'',
  action:()=>null,
  cssCont:'',
  cssBtn:'',
  tipoCart:false
}

ItemHeader.propTypes = {
  icono:PropTypes.element,
  title:PropTypes.string,
  action:PropTypes.func,
  cssCont:PropTypes.string,
  cssBtn:PropTypes.string,
  tipoCart:PropTypes.bool
};


export default ItemHeader;
