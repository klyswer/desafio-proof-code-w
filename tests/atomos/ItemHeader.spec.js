import React from 'react';
/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import ItemHeader from '../../src/front/atomos/itemHeader/ItemHeader';

describe('<ItemHeader />', () => {
  const title = "boton"
  const wrapper = shallow( <ItemHeader title={title} /> );

  test('Primer SnapShot: <ItemHeader />', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Caso prueba 1: validando render de props title:string.', () => {
    const itemEsperado = wrapper.find('.contenido_btn').text();
    expect(itemEsperado).toEqual(title);
  });

  test('Caso prueba 2: item de tipo boton.', () => {
    // const wrapper = shallow( <ItemHeader title={title} tipoCart handleClickCart={} /> );
    // const itemEsperado = wrapper.find('.contenido_btn').text();
    // expect(itemEsperado).toEqual(title);
  });

});