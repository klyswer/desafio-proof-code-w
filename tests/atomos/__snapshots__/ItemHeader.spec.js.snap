// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemHeader /> Primer SnapShot: <ItemHeader /> 1`] = `
<div
  className="nav-item "
>
  <button
    className="nav-link default_btn_header "
    onClick={[Function]}
    type="button"
  >
    <div
      className="contenido_btn"
    >
      <span />
      boton
    </div>
  </button>
</div>
`;
